kotlinpoet/src/wasmJsMain/kotlin/com/squareup/kotlinpoet/CodePoint.wasmJs.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+internal actual fun String.codePointAt(index: Int): CodePoint {
+  val str = this
+  val code = jsCodePointAt(str, index)
+  return CodePoint(code)
+}
+
+internal actual fun CodePoint.isLowerCase(): Boolean {
+  // TODO Will there be a problem? Is there a better way?
+  val code = this.code
+  val str = jsFromCodePoint(code)
+
+  if (str.length != 1) {
+    return false
+  }
+
+  return str.first().isLowerCase()
+}
+
+internal actual fun CodePoint.isUpperCase(): Boolean {
+  // TODO Will there be a problem? Is there a better way?
+  val code = this.code
+  val str = jsFromCodePoint(code)
+
+  if (str.length != 1) {
+    return false
+  }
+
+  return str.first().isUpperCase()
+}
+
+@Suppress("UNUSED_PARAMETER")
+private fun jsCodePointAt(str: String, index: Int): Int =
+  js("str.codePointAt(index)")
```

Comment:
Interesting that values returned by `js()` in `wasmJs` don't require casting same as in `js`, does WASM have its own flavour of `js()`?

Reply:
Yes, [Kotlin/JS's `js()`](https://kotlinlang.org/docs/js-interop.html) seems to be different from [Wasm's `js()`](https://kotlinlang.org/docs/wasm-js-interop.html#kotlin-functions-with-javascript-code) .